app/client/src/widgets/InputWidget.tsx

```
@@ -149,27 +161,35 @@ class InputWidget extends BaseWidget<InputWidgetProps, WidgetState> {
     if (this.props.maxChars) conditionalProps.maxChars = this.props.maxChars;
     if (this.props.maxNum) conditionalProps.maxNum = this.props.maxNum;
     if (this.props.minNum) conditionalProps.minNum = this.props.minNum;
+
+    const eventListeners = {
+      onKeyDown: this.handleKeyDown,
+    };
+
     return (
-      <InputComponent
-        value={value}
-        isInvalid={isInvalid}
-        onValueChange={this.onValueChange}
-        widgetId={this.props.widgetId}
-        inputType={this.props.inputType}
-        disabled={this.props.isDisabled}
-        label={this.props.label}
-        defaultValue={this.props.defaultText}
-        placeholder={this.props.placeholderText}
-        isLoading={this.props.isLoading}
-        multiline={
-          this.props.bottomRow - this.props.topRow > 1 &&
-          this.props.inputType === "TEXT"
-        }
-        stepSize={1}
-        onFocusChange={this.handleFocusChange}
-        showError={!!this.props.isFocused}
-        {...conditionalProps}
-      />
+      <div {...eventListeners}>
```

It is preferable to add this event listener directly on the input. There is a possibility in the future that someone could handle that event in the InputComponent itself and not let it bubble up